server/src/main/java/io/druid/server/coordination/SegmentChangeRequestDrop.java

```diff
@@ -62,4 +62,29 @@ public String toString()
            "segment=" + segment +
            '}';
   }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    SegmentChangeRequestDrop that = (SegmentChangeRequestDrop) o;
+
+    if (!segment.equals(that.segment)) {
```

Comment:
If the equality really is just on the segment, can the segment be acquired from a getter for this?